server-spi-private/src/main/java/org/keycloak/models/locking/GlobalLockProvider.java

```diff
@@ -17,57 +17,73 @@
 
 package org.keycloak.models.locking;
 
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.KeycloakSessionTaskWithResult;
 import org.keycloak.provider.Provider;
 
 import java.time.Duration;
 
 public interface GlobalLockProvider extends Provider {
 
+    class Constants {
+        public static final String KEYCLOAK_BOOT = "keycloak-boot";
+    }
+
     /**
-     * Effectively the same as {@code acquire(lockName, null)}
-     * <p />
-     * This method is intended to be used in a {@code try}-with-resources block.
+     * Effectively the same as {@code withLock(lockName, null, task)}
      *
      * @param lockName Identifier used for acquiring lock. Can be any non-null string.
-     * @return Instance of {@link GlobalLock} representing successfully acquired global lock.
+     * @param task The task that will be executed under the acquired lock
+     * @param <V> Type of object returned by the {@code task}
+     * @return Value returned by the {@code task}
      * @throws LockAcquiringTimeoutException When acquiring the global lock times out
-     *                          (see Javadoc of {@link #acquire(String, Duration)} for more details on how the time
-     *                          duration is determined)
-     * @throws NullPointerException When lockName is {@code null}.
+     *                                       (see Javadoc of {@link #withLock(String, Duration, KeycloakSessionTaskWithResult)} for more details on how the time
+     *                                       duration is determined)
+     * @throws NullPointerException          When lockName is {@code null}.
      */
-    default GlobalLock acquireLock(String lockName) throws LockAcquiringTimeoutException {
-        return acquire(lockName, null);
+    default <V> V withLock(String lockName, KeycloakSessionTaskWithResult<V> task) throws LockAcquiringTimeoutException {
+        return withLock(lockName, null, task);
     }
 
     /**
-     * Acquires a new global lock that is visible to all Keycloak nodes. The lock is non-reentrant.
-     * <p />
-     * The lock is guaranteed to be kept until the returned {@link GlobalLock} is closed
-     * using the {@link GlobalLock#close} method.
-     * <p />
-     * Some implementations may benefit from locks that are released at the end of transaction.
-     * For this purpose, the lifespan of the returned lock is limited by the transaction lifespan
-     * of the session which acquired this lock.
-     * <p />
-     * This method is intended to be used in a {@code try}-with-resources block.
-     * <p />
+     * Acquires a new non-reentrant global lock that is visible to all Keycloak nodes. If the lock was successfully
+     * acquired the method runs the {@code task} in a new transaction using method
+     * {@link org.keycloak.models.utils.KeycloakModelUtils#runJobInTransactionWithResult(KeycloakSessionFactory, KeycloakSessionTaskWithResult)}.
```

Comment:
Avoid going into the implementation details, summarize the behavior.

```suggestion
     * Acquires a new non-reentrant global lock that is visible to all Keycloak nodes. If the lock was successfully
     * acquired the method runs the {@code task} in a new transaction to ensure all data modified in is committed to the stores
     * before releasing the lock and returning to the caller.
```